src/Relude/List.hs

```diff
@@ -13,7 +13,34 @@ Utility functions to work with lists.
 module Relude.List
        ( module Relude.List.NonEmpty
        , module Relude.List.Reexport
+       , (!!?)
        ) where
 
+import Data.Bool (otherwise, (||))
+import Data.List (length, (!!))
+import Data.Maybe (Maybe (..))
```

Comment:
All such things should be exported from the corresponding `Relude` modules instead of `base` modules. This rule usually helps to make sure that `relude` contains enough things to implement their own stuff. Otherwise it looks a bit awkward when some standard things are imported from `base` and some are from `relude`